### ISSUE TITLE
Add a deprecation warning to `flax.training.lr_schedule`

### ISSUE THREAD

**Issue by IvyZX** [google/flax#2699]
The package should be deprecated in favor of Optax optimizer schedules: https://optax.readthedocs.io/en/latest/api.html#optimizer-schedules

It would help to add a warning in docstring or in code block for the existed flax learning rate schedules, to remind people to switch.

**Comment by 8bitmp3** [comment]
Since I raised this issue with `flax.training.lr_schedule` 😄 maybe I can start working on it if you don't have the bandwidth @IvyZX @marcvanzee LMKWYT

**Comment by IvyZX** [comment]
No worries - I just made a simple PR that could address this. Thank you for offering help!

**Comment by 8bitmp3** [comment]
:+1:
